### ISSUE TITLE
Not clear what attribute "id" means. Getting error 'Repository not found'

### ISSUE THREAD

**Issue by 99** [tomzo/gocd-json-config-plugin#10]
Hi, 
We're using this plugin very actively. Currently, we are using 17.10.0 gocd version, and it's not clear from the README what to add to the XML for the "id" attribute.
Example https://github.com/tomzo/gocd-json-config-example seems not updated too.

```<config-repos>
   <config-repo pluginId="json.config.plugin" id="repo1">
     <git url="https://github.com/tomzo/gocd-json-config-example.git" />
   </config-repo>
</config-repos>
...
<pipelines />
```
Becuase in some cases we are getting error
```: ERROR: Repository not found. STDERR: fatal: Could not read from remote repository. STDERR: STDERR: Please make sure you have the correct access rights STDERR: and the repository exists. ``` but git repo exists and working 

Thank you

**Comment by tomzo** [comment]
Hi,

`id` is arbitrary name of the configuration repository. It is needed for the GoCD API. see https://github.com/gocd/gocd/issues/3671

Regarding your error
> : ERROR: Repository not found. STDERR: fatal: Could not read from remote repository. STDERR: STDERR: Please make sure you have the correct access rights STDERR: and the repository exists. but git repo exists and working

That is unrelated to `id`. It is most likely missing authentication setup on GoCD server. Try to use the same repository as material in one of the pipelines, it will fail too with the same error.

**Comment by 99** [comment]
@tomzo I probably wasn't clear about id, where do I suppose to get it from? I saw the link you are referring to but it just has the same "id": "repo-1"? If I have a script which creates GH repo and adds the pipeline by modifying cruise-control.xml file. So before I didn't need any id to be added to the *xml file.   Where do I get "id" from?
I tried just adding id="repo1", id="repo2" everything started working.  It feels to me like this attribute doesn't mean anything 

Thank you
